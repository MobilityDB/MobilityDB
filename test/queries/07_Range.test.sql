-------------------------------------------------------------------------------
-- Tests for extensions of range data type.
-- File Range.c
-------------------------------------------------------------------------------

select intrange '[3,5)' << 5;
select 5 << intrange '[3,5)';

select intrange '[3,5)' >> 5;
select 5 >> intrange '[3,5)';

select intrange '[3,5)' &< 5;
select 5 &< intrange '[3,5)';

select intrange '[3,5)' &> 5;
select 5 &> intrange '[3,5)';

select intrange '[3,5)' -|- 5;
select 5 -|- intrange '[3,5)';

-------------------------------------------------------------------------------

select floatrange '[3.5, 5.5]' << 5.5;
select 5.5 << floatrange '[3.5, 5.5]';

select floatrange '[3.5, 5.5]' >> 5.5;
select 5.5 >> floatrange '[3.5, 5.5]';

select floatrange '[3.5, 5.5]' &< 5.5;
select 5.5 &< floatrange '[3.5, 5.5]';

select floatrange '[3.5, 5.5]' &> 5.5;
select 5.5 &> floatrange '[3.5, 5.5]';

select floatrange '[3.5, 5.5]' -|- 5.5;
select 5.5 -|- floatrange '[3.5, 5.5]';

-------------------------------------------------------------------------------
