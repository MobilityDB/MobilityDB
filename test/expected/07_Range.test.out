select intrange '[3,5)' << 5;
 ?column? 
----------
 t
(1 row)

select 5 << intrange '[3,5)';
 ?column? 
----------
 f
(1 row)

select intrange '[3,5)' >> 5;
 ?column? 
----------
 f
(1 row)

select 5 >> intrange '[3,5)';
 ?column? 
----------
 t
(1 row)

select intrange '[3,5)' &< 5;
 ?column? 
----------
 t
(1 row)

select 5 &< intrange '[3,5)';
 ?column? 
----------
 f
(1 row)

select intrange '[3,5)' &> 5;
 ?column? 
----------
 f
(1 row)

select 5 &> intrange '[3,5)';
 ?column? 
----------
 t
(1 row)

select intrange '[3,5)' -|- 5;
 ?column? 
----------
 t
(1 row)

select 5 -|- intrange '[3,5)';
 ?column? 
----------
 t
(1 row)

select floatrange '[3.5, 5.5]' << 5.5;
 ?column? 
----------
 f
(1 row)

select 5.5 << floatrange '[3.5, 5.5]';
 ?column? 
----------
 f
(1 row)

select floatrange '[3.5, 5.5]' >> 5.5;
 ?column? 
----------
 f
(1 row)

select 5.5 >> floatrange '[3.5, 5.5]';
 ?column? 
----------
 f
(1 row)

select floatrange '[3.5, 5.5]' &< 5.5;
 ?column? 
----------
 t
(1 row)

select 5.5 &< floatrange '[3.5, 5.5]';
 ?column? 
----------
 f
(1 row)

select floatrange '[3.5, 5.5]' &> 5.5;
 ?column? 
----------
 f
(1 row)

select 5.5 &> floatrange '[3.5, 5.5]';
 ?column? 
----------
 t
(1 row)

select floatrange '[3.5, 5.5]' -|- 5.5;
 ?column? 
----------
 f
(1 row)

select 5.5 -|- floatrange '[3.5, 5.5]';
 ?column? 
----------
 f
(1 row)

